samples/gltf_viewer.cpp

```diff
@@ -508,30 +508,64 @@ static void onClick(App& app, View* view, ImVec2 pos) {
     });
 }
 
-static utils::Path getPathForAsset(std::string_view string) {
+static utils::Path getPathForIBLAsset(std::string_view string) {
+    auto isIBL = [] (utils::Path file) -> bool {
+        return file.getExtension() == "ktx" || file.getExtension() == "hdr";
+    };
+
     utils::Path filename{ string };
     if (!filename.exists()) {
         std::cerr << "file " << filename << " not found!" << std::endl;
         return {};
     }
+
     if (filename.isDirectory()) {
-        auto files = filename.listContents();
-        for (const auto& file: files) {
-            if (file.getExtension() == "gltf" || file.getExtension() == "glb") {
+        bool found = false;
+        for (const auto& file: filename.listContents()) {
+            if (isIBL(file)) {
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            return {};
+        }
+    } else if (!isIBL(filename)) {
+        return {};
+    }
+
+    return filename;
+}
+
+static utils::Path getPathForGLTFAsset(std::string_view string) {
```

Comment:
I feel like this and `getPathForIBLAsset` are pretty much the same except for `isGLTF` and `isIBL`.  Maybe we can pull that into a paramter and just have a single method?

Reply:
Yeah I actually considered unifying them at first. Though it's possible to do that, it's quite tricky to distinguish the slightly different but critical behaviors. So it's likely to end up with convoluted code.